erdjs/src/proxyProvider.ts

```diff
@@ -16,155 +16,171 @@ const JSONbig = require("json-bigint");
+    return this.doPostGeneric("transaction/send", tx.toSendable(), (response) => new TransactionHash(response.txHash));
+  }
+
+  /**
+   * Simulates the processing of an already-signed {@link Transaction}.
+   */
+  async simulateTransaction(tx: Transaction): Promise<any> {
+    return this.doPostGeneric("transaction/simulate", tx.toSendable(), (response) => response);
+  }
+
+  /**
+   * Fetches the state of a {@link Transaction}.
+   */
+  async getTransaction(
+    txHash: TransactionHash,
+    hintSender?: Address,
+    withResults?: boolean
+  ): Promise<TransactionOnNetwork> {
+    let url = this.buildUrlWithQueryParameters(`transaction/${txHash.toString()}`, {
+      withSender: hintSender ? hintSender.bech32() : "",
+      withResults: withResults ? "true" : "",
+    });
+
+    return this.doGetGeneric(url, (response) => TransactionOnNetwork.fromHttpResponse(response));
+  }
+
+  /**
+   * Queries the status of a {@link Transaction}.
+   */
+  async getTransactionStatus(txHash: TransactionHash): Promise<TransactionStatus> {
+    return this.doGetGeneric(
+      `transaction/${txHash.toString()}/status`,
+      (response) => new TransactionStatus(response.status)
+    );
+  }
+
+  /**
+   * Fetches the Network configuration.
+   */
+  async getNetworkConfig(): Promise<NetworkConfig> {
+    return this.doGetGeneric("network/config", (response) => NetworkConfig.fromHttpResponse(response.config));
+  }
+
+  /**
+   * Fetches the network status configuration.
+   */
+  async getNetworkStatus(): Promise<NetworkStatus> {
+    return this.doGetGeneric("network/status/4294967295", (response) =>
+      NetworkStatus.fromHttpResponse(response.status)
+    );
+  }
+
+  /**
+   * Get method that receives the resource url and on callback the method used to map the response.
+   */
+  async doGetGeneric(resourceUrl: string, callback: (response: any) => any): Promise<any> {
+    let response = await this.doGet(resourceUrl);
+    return callback(response);
+  }
+
+  /**
+   * Post method that receives the resource url, the post payload and on callback the method used to map the response.
+   */
+  async doPostGeneric(resourceUrl: string, payload: any, callback: (response: any) => any): Promise<any> {
+    let response = await this.doPost(resourceUrl, payload);
+    return callback(response);
+  }
+
+  private async doGet(resourceUrl: string): Promise<any> {
+    try {
+      let url = `${this.url}/${resourceUrl}`;
+      let response = await axios.get(url, { timeout: this.timeoutLimit });
+      let payload = response.data.data;
+      return payload;
+    } catch (error) {
+      if (!error.response) {
+        Logger.warn(error);
+        throw new errors.ErrApiProviderGet(resourceUrl, error.toString(), error);
+      }
+
+      let errorData = error.response.data;
+      let originalErrorMessage = errorData.error || errorData.message || JSON.stringify(errorData);
+      throw new errors.ErrApiProviderGet(resourceUrl, originalErrorMessage, error);
```

Comment:
This is similar to `ApiProvider.handleApiError()`.

The comment at the top of the file says that we might merge `ProxyProvider` into `ApiProvider`. Maybe creating the function `ProxyProvider.handleApiError()`, just like in `ApiProvider`, will help with this merge.

Reply:
done